env/var.go

```diff
@@ -42,6 +43,27 @@ const (
 	DURATION
 )
 
+type FeatureStatus byte
+
+const (
+	Alpha FeatureStatus = iota
+	Beta
+	Stable
+	Unknown
+)
+
+func (s FeatureStatus) String() string {
+	switch s {
+	case Alpha:
+		return "Alpha"
+	case Beta:
+		return "Beta"
+	case Stable:
+		return "Stable"
```

Comment:
What does it mean for an env var to be stable? Don't we currently document that all env vars are alpha?

Put another way, why would we promote an env var to stable vs moving it to an API/mesh config?

Reply:
I am all for the idea of migrating all stable environment variables to an API, but that does not represent the current state of the project.  For instance, I believe we set the PILOT_TRACE_SAMPLING variable on every istio deployment, and it should probably be considered stable.

Reply:
PILOT_TRACE_SAMPLING will be deprecated in 1.9 by the tracing api. We shouldn't mark something "stable" just because its been around for a while, we should mark it stable if we know its the right long term API and its properly tested

Reply:
The overall idea here is to give users a way to visualize the maturity of the features they are leveraging.  We've already got that for Annotations with Nate's PR, and this would provide visibility into env vars, which could be combined in a single istioctl command to show users what features they are using, and what support they can expect.  Given the prominence of many env vars, I doubt we would feed comfortable telling our users that all of them have only alpha level support.

Reply:
Can we track the same state we're now tracking for annotation to cover the transition intent

deprecated, deprecatedBy

etc.

Reply:
Yes.  It might be helpful to track the removal of deprecated features too, so we can highlight that when checking upgrade readiness.

It's also useful to note that users may not be setting these directly, but via operator or helm values.

Reply:
Should we add experimental to here as well?